engine/execution/ingestion/core.go

```
@@ -88,20 +103,54 @@ func NewCore(
 		collectionFetcher: collectionFetcher,
 		eventConsumer:     eventConsumer,
 	}
-}
 
-func (e *Core) Ready() <-chan struct{} {
-	if e.stopControl.IsExecutionStopped() {
-		return e.unit.Ready()
+	err := e.throttle.Init(e.processables)
+	if err != nil {
+		e.log.Fatal().Err(err).Msg("fail to initialize throttle engine")
```

Can we instead return this error?